### PR TITLE
No longer use deprecated Tyxml printer

### PR DESCRIPTION
**Pull request by vouillon**

### CODE REVIEW

File: src/http/ocsigen_senders.ml [hunk 1]
```diff
@@ -65,7 +64,9 @@ module Make_XML_Content(Xml : Xml_sigs.Iterable)
         Typed_xml.Info.alternative_content_types
         Typed_xml.Info.content_type in
     let encode x = fst (Xml_print.Utf8.normalize_html x) in
-    let x = Htmlprinter.print ~encode ~advert c in
+    let x =
+      Ocsigen_stream.of_string
+        (Format.asprintf "%a" (Htmlprinter.pp ~encode ~advert ()) c) in
```
Comment (Drup) [inline src/http/ocsigen_senders.ml]: That induces an extra copying. That's why we avoided making the change before. :/

Comment (vouillon) [inline src/http/ocsigen_senders.ml]: Is it really worse than the deprecated printer, that performs a _huge_ amount of allocations?

Comment (Drup) [inline src/http/ocsigen_senders.ml]: Ah, right, maybe not. :p

**Comment by vasilisp** [comment]
As far as I am concerned, you can merge this. For info, the Cohttp version doesn't have `Ocsigen_senders` at all (and Eliom uses its own non-deprecated printers for producing the response).

**Pull request merged**